benchflow-dsl/src/main/scala/cloud/benchflow/dsl/ExplorationSpace.scala

```diff
@@ -52,46 +51,83 @@ object ExplorationSpace {
   /**
    *
    * @param explorationSpace         the exploration space where the experiment is
-   * @param experimentNumber         the number of the experiment (index)
+   * @param experimentIndex          the index of the experiment
    * @param testDefinitionYamlString the test definition to build the experiment definition from as a YAML string
    * @param dockerComposeYamlString  the docker compose to build the experiment docker-compose from as a YAML string
    * @return a tuple of experiment definition YAML string and docker compose YAML string
    */
   def generateExperimentBundle(
-    explorationSpace: ExplorationSpaceDimensions,
-    experimentNumber: ExperimentNumber,
+    explorationSpace: ExplorationSpace,
+    experimentIndex: ExperimentIndex,
     testDefinitionYamlString: String,
     dockerComposeYamlString: String): (ExperimentDefinitionYamlString, DockerComposeYamlString) = {
 
-    // get the experiment configuration from the exploration space
-
-    // generate the experiment definition YAML
-
-    // generate the docker compose YAML
-    ("to do", "to do")
+    // build the experiment definition
+    var experimentBuilder = BenchFlowDSL.experimentYamlBuilderFromTestYaml(testDefinitionYamlString)
+
+    experimentBuilder = explorationSpace.usersDimension match {
+      case Some(list) => experimentBuilder.numUsers(list(experimentIndex))
+      case None => experimentBuilder
+    }
+
+    val experimentDefinitionYamlString = experimentBuilder.build()
+
+    // build the docker compose YAML
+    var dockerComposeBuilder = BenchFlowDSL.dockerComposeYamlBuilderFromDockerComposeYaml(dockerComposeYamlString)
+
+    explorationSpace.memoryDimension match {
+      case Some(map) => map foreach {
+        case (serviceName, list) => dockerComposeBuilder = dockerComposeBuilder.memLimit(serviceName, list(experimentIndex))
+      }
+      case None => // nothing to do
+    }
+
+    explorationSpace.environmentDimension match {
+      case Some(serviceMap) => serviceMap foreach {
+        case (serviceName, environmentMap) => environmentMap foreach {
+          case (environmentVariable, list) => dockerComposeBuilder = dockerComposeBuilder.environmentVariable(
+            serviceName,
+            environmentVariable,
+            list(experimentIndex))
+        }
+      }
+      case None => // nothing to do
+    }
+
+    val newDockerComposeYamlString = dockerComposeBuilder.build()
+
+    (experimentDefinitionYamlString, newDockerComposeYamlString)
   }
 
   /**
+   * Generates and experiment bundle from the given exploration space point. Assumes the point exists in the
```

Comment:
I just saw this was an old commit. It actually returns an option, e.g., `None` if the point doesn't exist (or more than one). I will update the JavaDoc.